### ISSUE TITLE
provision.cleanup.incomplete.projects lets tests/make test fail

### ISSUE THREAD

**Issue by renedupont** [opendevstack/ods-core#740]
**Describe the bug**

When running `ods-core/tests>make test` more than once against our ODS Installation it fails because it seems that while
[provision.cleanup.incomplete.projects](https://github.com/opendevstack/ods-core/blob/master/ods-provisioning-app/ocp-config/cm.yml#L225) has been set to false by default (see issue https://github.com/opendevstack/ods-provisioning-app/issues/507), it needs to be set to true to delete that ODSVERIFY project that gets created in TestVerifyOdsProjectProvisionThruProvisionApi. If it is not set to true you'll receive in the very first run:

```
delete / get project - DELETE
curl request successful...

... displaying HTTP response body (content from './response.txt'):
{"timestamp":"2020-07-31T12:54:37.828+0000","status":500,"error":"Internal Server Error","message":"Cleanup of projects is NOT allowed","path":"/api/v2/project/ODSVERIFY"}
... displaying HTTP response code
http_resp_code=500

something went wrong... endpoint responded with error code [HTTP CODE=500] (expected was 200)
```

which is ok I guess since it does not fail the test run and creates the ODSVERIFY project afterwards and carries on. But once you run `make test` again, the test run will fail due to:

```
        curl request successful...

         ... displaying HTTP response body (content from './response.txt'):
        Project with key (ODSVERIFY) or name (ODSVERIFY) already exists
        ... displaying HTTP response code
        http_resp_code=500
 
        something went wrong... endpoint responded with error code [HTTP CODE=500] (expected was 200)
```

Now, if I just change that flag to true in my application.properties of the provApp, it will delete the project and TestVerifyOdsProjectProvisionThruProvisionApi will succeed, but I will not pass TestVerifyOdsNamespaceProvAppOcpConfig...

To summarize: If I run `make test` more than once and don't change the default project key ODSVERIFY it will fail the verify phase. I cheated my way around it by deleting the project 'manually' to pass the verify phase but that can't be the answer.

I would also question the use of provision.cleanup.incomplete.projects in that test scenario here, I think that flag was introduced for something else.

fyi @clemensutschig @georgfedermann @michaelsauter

**Comment by clemensutschig** [comment]
This is a precondition for the tests.. will add to docs ...

Ill try o find out whats up with https://github.com/opendevstack/ods-provisioning-app/issues/351

**Comment by clemensutschig** [comment]
@georgfedermann

**Comment by renedupont** [comment]
yes very good. I changed provision.cleanup.incomplete.projects locally in ods-configuration/ods-core.env to true and in the ods namespace aswell and the verify test passed.

I still wonder though if provision.cleanup.incomplete.projects should be in the end the flag to decide whether or not test projects should be deleted in the test run. I find it confusing because it says cleanup.incomplete.projects and not cleanup.test.projects - or are test projects always incomplete by definition? Was this flag initially introduced for tests?

**Comment by clemensutschig** [comment]
@renedupont - this was introduced to prevent unwanted cleanups ... and then sort of misused to disable the whole feature :(
